logstash-core/src/test/java/org/logstash/instrument/metrics/timer/TimerMetricTest.java

```diff
@@ -0,0 +1,175 @@
+package org.logstash.instrument.metrics.timer;
+
+import com.google.common.util.concurrent.Monitor;
+import org.junit.Test;
+import org.logstash.instrument.metrics.ManualAdvanceClock;
+
+import java.time.Duration;
+import java.util.Objects;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertThat;
+
+/**
+ * This {@code TimerMetricTest} is meant to be inherited by tests covering the implementations of {@link TimerMetric},
+ * and includes baseline tests for guaranteeing that the value of the metric includes the duration of
+ * execution before control is given back to the caller.
+ *
+ * <p>At an <em>interface</em>-level, we make no guarantees that tracked executions will be committed into the
+ * value <em>until</em> control is returned, which means that the value may or may not include "uncommitted" or
+ * currently-executing tracked timings. This means that our assertions can only apply when there are no concurrent
+ * executions happening.
+ */
+public abstract class TimerMetricTest {
```

Comment:
Huge thanks for the clear test cases, especially blocked timed task tests are super helpful.